openstf/tasks/utils/predictionjobloop.py

```diff
@@ -106,6 +107,7 @@ def map(self, function, *args, **kwargs):
         """
         pids_successful = []
         pids_unsuccessful = []
+        pids_unsuccessful_dict = defaultdict(list)
```

Comment:
Perhaps we can keep it simpler by only using one data structure for unsuccessful pid's. Now both contain the pid's. I would suggest to use `pid_unsucessful` or `pid_exceptions` as a normal dicts:

```python
pids_unsucessful = {}
```

Or perhaps I'm missing why a defaultdict is required here?

Reply:
defaultdict(list) is very practical, as it allows easy 'append to a list if a key is already present, or create the list for that key if the key is not yet present'-functionality.